docs: add svg template example (#30559)
angular/angular: aio/content/guide/template-syntax.md

</code-example>

The `$any()` cast function works anywhere in a binding expression where a method call is valid.

## SVG in templates

It is possible to use SVG as valid templates in Angular. All of the template syntax below is applicable to both SVG and HTML.
Learn more in the SVG [1.1](https://www.w3.org/TR/SVG11/) and [2.0](https://www.w3.org/TR/SVG2/) specifications.

> Why would you use an .svg as template, instead of simply adding it as image to your application?

Reviewer: > use an .svg as template

This sounds a little odd to me. How about: `use SVG as template` or ``use an `.svg` file as template``

Reviewer 2: Done.

> When you use an SVG as the template, you are be able to use directives and bindings just like with HTML templates. This means that you will be able to dynamically generate interactive graphics.

Reviewer: Typo: _you are be able_ --> _you are able_

Reviewer 2: done.

Refer to the sample code snippet below for a syntax example:

> <code-example path="template-syntax/src/app/svg.component.ts" header="src/app/app.component.ts">

Reviewer: Why not name it `src/app/svg.component.ts` (keeping in-line with the recommendation to match component name and file name)?
Same for the template file below (`app.component.svg` --> `svg.component.svg`).

Reviewer 2: will change.

</code-example>

The `$any()` cast function works anywhere in a binding expression where a method call is valid.

## SVG in templates

> Scalable Vector Graphics (SVG) is an XML-based vector image format for two-dimensional graphics with support for interactivity and animation.

Reviewer 3: It may be helpful to link to the [1.1 spec](https://www.w3.org/TR/SVG11/) and [2.0 spec](https://www.w3.org/TR/SVG2/) here.

Reviewer 2: done.

> You can use the SVG as valid template in Angular, all the template syntax which you are going to learn in next sections are applicable to SVG as well.

Reviewer 3: ```suggestion
You can use SVG as valid templates in Angular. All of the template syntax below is applicable to both SVG and HTML.
```

Reviewer 2: Done.

Why would you use an .svg as template, instead of simply adding it as image to your application?

> When you use an svg as template, you will be able to use directives and bindings just like with html templates. That means you will be able to dynamically generate interactive graphics.

Reviewer 3: ```suggestion
When you use SVG as the template, you are able to use directives and bindings just like HTML templates. This means that you will be able to dynamically generate interactive graphics.
```

Reviewer 2: Done.

> Refer the sample code snippet below for syntax:

Reviewer 3: ```suggestion
Refer to the code snippet below for a syntax example:
```

Reviewer 2: Done.

<code-example path="template-syntax/src/app/svg.component.ts" header="src/app/app.component.ts">
</code-example>

Add the below code to your `app.component.svg` file:

```html
<svg>
	<g>
		<rect x="0" y="0" width="100" height="100"
		 [attr.fill]="fill" (click)="changeColor()" />
		<text x="120" y="50">click the rectangle to change the fill color</text>
</g>
</svg>

> ```

Reviewer: This code snippet should [come from a docregion][1], not  ahrd-coded example.

[1]: https://angular.io/guide/docs-style-guide#code-snippet-from-a-file

Reviewer 2: The problem is code-example is not rendering SVG tags properly.

Reviewer: Aha! That sounds like a bug 😕
Can you, please, update the PR to use `docregion`/`code-example` and ping me to look into why it is not working as expected?

Reviewer 2: I have updated the PR, it is getting failed. Please check.

> In above code you can see we are using `click()` event binding and `[attr.fill]="fill"` property binding syntax.

Reviewer 3: ```suggestion
Here you can see the use of a `click()` event binding and the property binding syntax (`[attr.fill]="fill"`).
```
It might be helpful to add one more brief sentence to just describe how this would behave (i.e. clicking will change the image's color).

Reviewer 2: Done.

</code-example>

The `$any()` cast function works anywhere in a binding expression where a method call is valid.

## SVG in templates

Scalable Vector Graphics (SVG) is an XML-based vector image format for two-dimensional graphics with support for interactivity and animation.
You can use SVG as valid templates in Angular. All of the template syntax below is applicable to both SVG and HTML.

> You can read SVG [1.1 Spec](https://www.w3.org/TR/SVG11/) and [2.0 Spec](https://www.w3.org/TR/SVG2/).

Reviewer 3: ```suggestion
Learn more in the SVG [1.1](https://www.w3.org/TR/SVG11/) and [2.0](https://www.w3.org/TR/SVG2/) specifications.
```

Reviewer 2: done.
